src/libcore/iter/adapters/mod.rs

```diff
@@ -1509,6 +1509,17 @@ impl<I> DoubleEndedIterator for Skip<I> where I: DoubleEndedIterator + ExactSize
         }
     }
 
+    #[inline]
+    fn nth_back(&mut self, n: usize) -> Option<I::Item> {
+        if n < self.len() {
+            self.iter.nth_back(n)
+        } else {
```

Comment:
In case `n >= self.len() && self.len() != 0`, you need to do `let _ = self.iter.nth_back(self.len() - 1);` (or whatever that takes `self.len()`-items from the back) since the underlying iterator could have side effects.

```rust
let mut v = vec![];
let mut it = [1, 2, 3, 4, 5]
    .iter()
    .cloned()
    .inspect(|x| v.push(*x))
    .skip(1);
assert_eq!(it.len(), 4);
assert_eq!(it.nth_back(4), None);
assert_eq!(v, vec![5, 4, 3, 2]);
```

Reply:
Why `let _ = ` and not just the call to `nth_back`?

Reply:
It was just for expressing intent to throw away the return value of `nth_back`, please ignore it.